core/src/apps/bitcoin/sign_tx/decred.py

```diff
@@ -423,3 +423,21 @@ def write_tx_input_witness(
             pubkey,
             signature,
         )
+
+    # scriptPubKey derivation
+    # ===
+
+    def input_derive_script(
+        self, txi: TxInput, node: bip32.HDNode | None = None
+    ) -> bytes:
+        if input_is_external(txi):
+            assert txi.script_pubkey is not None  # checked in _sanitize_tx_input
+            return txi.script_pubkey
+
+        if node is None:
+            node = self.keychain.derive(txi.address_n)
+
+        address = addresses.get_address(txi.script_type, self.coin, node, txi.multisig)
+        return scripts_decred.output_derive_script(
```

Comment:
silence the type checker:
```suggestion
        assert txi.decred_tree is not None and txi.decred_staking_spend is not None
        return scripts_decred.output_derive_script(
```